caikit/core/data_model/base.py

```diff
@@ -52,6 +52,8 @@
 
 
 class _DataBaseMetaClass(type):
+    fields: Tuple
```

Comment:
I think there are a few more to add here based on all of the additions to `atters` [below](https://github.com/caikit/caikit/pull/617/files#diff-69f5427572b1727bc0dd45f16302745f3926d21ce72d5d63eaa4391b23269a9eR108)

(also, this would be really nice to have explicitly defined, at least for the public class attributes!)